Sources/NIOSSL/SSLCallbacks.swift

```diff
@@ -173,6 +173,84 @@ public struct PSKClientIdentityResponse: Sendable {
     }
 }
 
+public struct NIOSSLClientExtensionValues: Hashable {
+    public var serverHostname: String?
+
+    public init(serverHostname: String?) {
+        self.serverHostname = serverHostname
+    }
+}
+
+/// A callback that can used to support multiple or dynamic TLS hosts.
+///
+/// When set, this callback will be invoked once per TLS hello. The provided `NIOSSLClientExtensionValues` will contain the
+/// host name indicated in the TLS client hello, while the provided `NIOSSLContext` will be the current
+/// server context being used for the handshake.
+///
+/// Within this callback, the user can create and return a new server `NIOSSLContext` for the given host,
+/// return the same context provided if it is sufficient to complete the handshake, or return `throw` in the event
+/// of an exception.
+///
+public typealias NIOSSLContextCallback = @Sendable (NIOSSLClientExtensionValues, NIOSSLContext) -> EventLoopPromise<NIOSSLContext>
```

Comment:
This function should accept a promise as an argument, not return one.

Reply:
@Lukasa Do you mind expanding on this? Since the initial PR this callback was meant to be a user-land callback where the user asynchronously provides us a new context. How could the argument be a future promise?

Reply:
Is the idea that we create the promise and then in user code they can fulfill it?